src/app/shared/services/helper.service.ts

```diff
@@ -128,14 +96,14 @@ export class HelperService {
     let index: number = 0;
     let parentMap: any[] = [];
 
-    let showingId = this.getCheckpointOrStorageId(report, true);
+    let showingId: string = this.getCheckpointOrStorageId(report, true);
```

Comment:
let can be a const
parentMap can also be a const
Also this function should be refactored, so I dont understand why you've added types here

Reply:
I added the change.

> Also this function should be refactored, so I dont understand why you've added types here

I was just going through the helper service adding types wherever it was easy to add it so we don't have to later on.